snuba/query/processors/uniq_in_select_and_having.py

```diff
@@ -0,0 +1,92 @@
+"""This is a hacky query processor put in for the purpose of
+mitigating clickhouse crashes. When a query is sent to clickhouse
+which has a `uniq` in the `HAVING` clause but not in the `SELECT`
+clause, the query node will crash. This may be fixed with future versions
+of clickhouse but we have not upgraded yet. This exists as a check so that we
+return an exception to the user but don't crash the clickhouse query node
+"""
+
+import logging
+from dataclasses import fields
+from typing import List, Sequence
+
+from snuba.clickhouse.processors import QueryProcessor
+from snuba.clickhouse.query import Query
+from snuba.query.exceptions import InvalidQueryException
+from snuba.query.expressions import Expression, FunctionCall, NoopVisitor
+from snuba.request.request_settings import RequestSettings
+from snuba.state import get_config
+
+
+class MismatchedAggregationException(InvalidQueryException):
+    pass
+
+
+class _FunctionNameFindingVisitor(NoopVisitor):
+    def __init__(self, function_name_to_find: str):
+        self.function_name_to_find = function_name_to_find
+        self.found_functions: List[FunctionCall] = []
+
+    def visit_function_call(self, exp: FunctionCall) -> None:
+        for param in exp.parameters:
+            param.accept(self)
+        if exp.function_name == self.function_name_to_find:
+            self.found_functions.append(exp)
+        return None
+
+
+def _expressions_equal_ignore_aliases(exp1: Expression, exp2: Expression) -> bool:
+    if type(exp1) != type(exp2):
+        return False
+    fields_equal = True
+    for field in fields(exp1):
+        if field.name == "alias":
+            continue
+        field_val1 = getattr(exp1, field.name)
+        field_val2 = getattr(exp2, field.name)
+        if isinstance(field_val1, Expression):
+            fields_equal &= _expressions_equal_ignore_aliases(field_val1, field_val2)
+        else:
+            fields_equal &= field_val1 == field_val2
```

Comment:
Not sure this works for the parameters of a function:
`FunctionCall` has three fields:
```
alias: string
name: string
parameters: Tuple(expression....)
```
When you check for the equality if the parameters you will fall to line 50, where the two tuples will be compared to each other. And the elements of the tuples will be compared with `==` which takes the alias into account.

Reply:
Good catch! What do you think of me adding this functionality to the Expression class?

Reply:
Do you mean equality without aliases ? If yes, let me think if there was any reason for not doing it.

Reply:
yes that is what I mean. Functional equivalency

Reply:
Well the idea was to solve this very problem with matchers, but the use case of taking an expression and find an identical one was not very common. I have nothing against adding it to the expression. Maybe you can build something more flexible by generating the matcher from the expression, so that the result can be used to build a larger matcher, etc.

Reply:
ok let me give the matcher path a shot

Reply:
Actually, while I think this is a cool thing to build. I think it may be too much overhead for testing it at this time given that this is an active production issue. There's also nothing preventing us from implementing the matcher solution later and swapping the functional equivalency for it.